tqdm/_tqdm.py

```diff
@@ -470,14 +476,11 @@ def _decr_instances(cls, instance):
                 for inst in cls._instances:
                     if inst.pos > instance.pos:
                         inst.pos -= 1
-            # Kill monitor if no instances are left
-            if not cls._instances and cls.monitor:
-                cls.monitor.exit()
-                try:
+                # Kill monitor if no instances are left
+                if not cls._instances and cls.monitor:
+                    cls.monitor.exit()
                     del cls.monitor
-                except AttributeError:
-                    pass
```

Comment:
@casperdcl Why remove this exception catch? It cannot happen anymore because of the lock? But since the lock is optional, I think the AttributeError might still happen when lock is disabled...

Reply:
I thought you had removed this (that's what it looked like when I was rebasing). Happy to re-add.

Reply:
Also, is the lock optional? There's a default global lock which means `tqdm` is now thread safe by default. You don't need to `set_lock`...

Reply:
I don't remember removing the exception check, and from the commit history it's not part of my commits :-/

About the lock being optional, yes it is, because I think there is no way to provide default global lock on Windows, the user needs to provide a lock from the parent level. Default global lock only works on Linux/Mac, but unluckily not on platforms not supporting fork()...

Reply:
BTW that's why I added an argument to provide tqdm with a user defined lock, because this is necessary for Windows, else we could just use only a boolean argument "enable_lock", but for Windows (and platforms not supporting fork()), we also need to provide user with an argument to supply a user defined lock (which must be defined in the script at the parent level to tqdm, in other words in the calling script).

Reply:
Also I figured that providing user the ability to supply one's own lock could be useful in very specific applications or platforms where the locking mechanism might be different, this way we foolproof tqdm to work in any kind of concurrent environment as long as the user can provide a lock that works for this environment.